vulnerabilities/importer.py

```diff
@@ -194,6 +194,64 @@ def from_url(cls, url):
         return cls(url=url)
 
 
+@dataclasses.dataclass(eq=True)
+@functools.total_ordering
+class CodeCommitData:
+    commit_hash: str
+    vcs_url: str
+
+    commit_author: Optional[str] = None
+    commit_message: Optional[str] = None
+    commit_date: Optional[datetime.datetime] = None
+
+    def __post_init__(self):
+        if not self.commit_hash:
+            raise ValueError("Commit must have a non-empty commit_hash.")
+
+        if not is_commit(self.commit_hash):
+            raise ValueError("Commit must be a valid a commit_hash.")
+
+        if not self.vcs_url:
+            raise ValueError("Commit must have a non-empty vcs_url.")
+
+    def __lt__(self, other):
+        if not isinstance(other, CodeCommitData):
+            return NotImplemented
+        return self._cmp_key() < other._cmp_key()
+
+    # TODO: Add cache
+    def _cmp_key(self):
+        return (
+            self.commit_hash,
+            self.vcs_url,
+            self.commit_author,
+            self.commit_message,
+            self.commit_date,
+        )
+
+    def to_dict(self) -> dict:
+        """Return a normalized dictionary representation of the commit."""
+        return {
+            "commit_hash": self.commit_hash,
+            "vcs_url": self.vcs_url,
+            "commit_author": self.commit_author,
+            "commit_message": self.commit_message,
+            "commit_date": self.commit_date,
+        }
+
+    @classmethod
+    def from_dict(cls, data: dict):
+        """Create a Commit instance from a dictionary."""
+        commit_date = data.get("commit_date")
+        return cls(
+            commit_hash=str(data.get("commit_hash", "")),
```

Comment:
`commit_hash` is always string, right?

Reply:
@ziadhany there is no point of creating commits with empty strings. If commit hash is not present then let's not create a commit at all.

Reply:
This is still not fixed. Please check this! Thanks!